tests/ga/test_extension.py

```diff
@@ -548,6 +548,48 @@ def _assert_handler_status_and_manifest_download_count(protocol, test_data, mani
             # The extension manifest should be downloaded twice now as incarnation changed once
             _assert_handler_status_and_manifest_download_count(protocol, test_data, 2)
 
+    def test_it_should_fail_handler_on_bad_extension_config_and_report_error(self, mock_get, mock_crypt_util, *args):
+
+        invalid_config_dir = os.path.join(data_dir, "wire", "invalid_config")
+        self.assertGreater(len(os.listdir(invalid_config_dir)), 0, "Not even a single bad config file found")
+
+        for bad_config_file_path in os.listdir(invalid_config_dir):
+            bad_conf = DATA_FILE.copy()
+            bad_conf["ext_conf"] = os.path.join(invalid_config_dir, bad_config_file_path)
+            test_data = mockwiredata.WireProtocolData(bad_conf)
+            exthandlers_handler, protocol = self._create_mock(test_data, mock_get, mock_crypt_util, *args)
+
+            with patch('azurelinuxagent.common.event.add_event') as patch_add_event:
+                exthandlers_handler.run()
+                self._assert_handler_status(protocol.report_vm_status, "NotReady", 0, "1.0.0")
+
+                invalid_config_errors = [kw for _, kw in patch_add_event.call_args_list if
+                                         kw['op'] == WALAEventOperation.InvalidExtensionConfig]
+                self.assertEqual(1, len(invalid_config_errors), "Error not logged and reported to Kusto")
+
+    def test_it_should_process_valid_extensions_if_present(self, mock_get, mock_crypt_util, *args):
```

Comment:
Is this behavior what @narrieta is talking about up above (in `.../goal_state.py`)? I.e. he wanted a single invalid extension to fail the entire goal state?

Reply:
No not really, this test ensures that even if some extension's config is bad, we still process other extensions. The ask was to fail all extensions instead. 

We're not going with that behavior though, we're keeping this behavior